cmake/CMakeLists.txt

```diff
@@ -139,6 +139,7 @@ option(onnxruntime_DISABLE_ML_OPS "Disable traditional ML ops" OFF)
 option(onnxruntime_DISABLE_SPARSE_TENSORS "Disable sparse tensors data types" OFF)
 option(onnxruntime_DISABLE_OPTIONAL_TYPE "Disable optional type" OFF)
 option(onnxruntime_DISABLE_FLOAT8_TYPES "Disable float 8 types" OFF)
+option(onnxruntime_DISABLE_FLOAT16_TYPES "Disable float 16 types" OFF)
```

Comment:
float16 one seems a little misleading if it's only affecting part of MLAS given float16 is used in a lot of other places and we're not disabling any of those. 

what's the binary size saving of having this?

Reply:
> the float16 one seems a little misleading if it's only affecting part of MLAS given float16 is used in a lot of other places and we're not disabling any of those.
> 
> what's the binary size saving of having this?

hm, okay that's true. excluding the fp16 support files in total saves about 20KB uncompressed.

how about using another name e.g. DISABLE_PARTIAL_MLAS_FLOAT16_TYPES?/better suggestions?

Reply:
Possibly getting too specific for a top-level option in the ORT build.

Maybe we can add this to the list of potential reductions and do it in a more complete way if needed, as there's float16 as well as bfloat16 to consider.

I probably wouldn't exclude it for all minimal builds either as there's no way to enable it in a minimal build if you wanted to.

Reply:
cool, updated.